spot_driver/spot_driver/spot_ros2.py

```diff
@@ -1466,245 +1470,216 @@ def handle_max_vel(self, request: SetVelocity.Request, response: SetVelocity.Res
+        if feedback.status.value != feedback.status.STATUS_PROCESSING:
             return GoalResponse.IN_PROGRESS
 
-        if feedback.command.command_choice == feedback.command.COMMAND_FULL_BODY_FEEDBACK_SET:
-            full_body_feedback = feedback.command.full_body_feedback
-            if full_body_feedback.status.value != full_body_feedback.status.STATUS_PROCESSING:
-                return GoalResponse.IN_PROGRESS
-            if full_body_feedback.feedback.feedback_choice == full_body_feedback.feedback.FEEDBACK_STOP_FEEDBACK_SET:
+        fb = feedback.feedback
+        choice = fb.feedback_choice
+
+        if choice == fb.FEEDBACK_STOP_FEEDBACK_SET:
+            return GoalResponse.SUCCESS
+        elif choice == fb.FEEDBACK_FREEZE_FEEDBACK_SET:
+            return GoalResponse.SUCCESS
+        elif choice == fb.FEEDBACK_SELFRIGHT_FEEDBACK_SET:
+            if fb.selfright_feedback.status.value == fb.selfright_feedback.status.STATUS_COMPLETED:
                 return GoalResponse.SUCCESS
-            elif (
-                full_body_feedback.feedback.feedback_choice == full_body_feedback.feedback.FEEDBACK_FREEZE_FEEDBACK_SET
-            ):
+            return GoalResponse.IN_PROGRESS
+
+        elif choice == fb.FEEDBACK_SAFE_POWER_OFF_FEEDBACK_SET:
+            if fb.safe_power_off_feedback.status.value == fb.safe_power_off_feedback.status.STATUS_POWERED_OFF:
                 return GoalResponse.SUCCESS
-            elif (
-                full_body_feedback.feedback.feedback_choice
-                == full_body_feedback.feedback.FEEDBACK_SELFRIGHT_FEEDBACK_SET
+            return GoalResponse.IN_PROGRESS
+
+        elif choice == fb.FEEDBACK_BATTERY_CHANGE_POSE_FEEDBACK_SET:
+            if fb.battery_change_pose_feedback.status.value == fb.battery_change_pose_feedback.status.STATUS_COMPLETED:
+                return GoalResponse.SUCCESS
+            if fb.battery_change_pose_feedback.status.value == fb.battery_change_pose_feedback.status.STATUS_FAILED:
+                return GoalResponse.FAILED
+            return GoalResponse.IN_PROGRESS
+
+        elif choice == fb.FEEDBACK_PAYLOAD_ESTIMATION_FEEDBACK_SET:
+            if fb.payload_estimation_feedback.status.value == fb.payload_estimation_feedback.status.STATUS_COMPLETED:
+                return GoalResponse.SUCCESS
+            if fb.payload_estimation_feedback.status.value == fb.payload_estimation_feedback.status.STATUS_SMALL_MASS:
+                return GoalResponse.SUCCESS
+            if fb.payload_estimation_feedback.status.value == fb.payload_estimation_feedback.status.STATUS_ERROR:
+                return GoalResponse.FAILED
+            return GoalResponse.IN_PROGRESS
+        elif choice == fb.FEEDBACK_CONSTRAINED_MANIPULATION_FEEDBACK_SET:
+            if (
+                fb.constrained_manipulation_feedback.status.value
+                == fb.constrained_manipulation_feedback.status.STATUS_RUNNING
```

Comment:
Does constrained manipulation never return SUCCESS?

Reply:
@abaker-bdai

Reply:
https://github.com/bdaiinstitute/bosdyn_msgs/blob/main/msg/ConstrainedManipulationCommandFeedbackStatus.msg Looking at the defined status values, it appears that it shouldn't return success. The status values imply either that it's running or it has failed to me